code/modules/projectiles/guns/misc/chem_gun.dm

```diff
@@ -13,21 +13,82 @@
 	custom_materials = list(/datum/material/iron=2000)
 	clumsy_check = FALSE
 	fire_sound = 'sound/items/syringeproj.ogg'
-	var/time_per_syringe = 300
-	var/syringes_left = 5
-	var/max_syringes = 5
+	var/time_per_syringe = 200
+	var/syringes_left = 3
+	var/max_syringes = 6
 	var/last_synth = 0
+	var/obj/item/reagent_containers/glass/bottle/vial/vial
+	var/list/allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/small, /obj/item/reagent_containers/glass/bottle/vial/large)
+	var/quickload = TRUE //
 
 /obj/item/gun/chem/Initialize()
 	. = ..()
 	chambered = new /obj/item/ammo_casing/chemgun(src)
 	START_PROCESSING(SSobj, src)
-	create_reagents(100, OPENCONTAINER)
 
 /obj/item/gun/chem/Destroy()
 	. = ..()
 	STOP_PROCESSING(SSobj, src)
 
+//bunch of hypospray copy paste
+/obj/item/gun/chem/examine(mob/user)
+	. = ..()
+	if(vial)
+		. += "[vial] has [vial.reagents.total_volume]u remaining."
+	else
+		. += "It has no vial loaded in."
+
+/obj/item/gun/chem/proc/unload_hypo(obj/item/I, mob/user)
+	if((istype(I, /obj/item/reagent_containers/glass/bottle/vial)))
+		var/obj/item/reagent_containers/glass/bottle/vial/V = I
```

Comment:
Technically not necessary, since you're not actually using any properties of V that I doesn't have

Reply:
Going to leave it for now.